packages/apptive_grid_core/lib/src/network/apptive_grid_client.dart

```diff
@@ -346,26 +339,15 @@ class ApptiveGridClient extends ChangeNotifier {
     Map<String, String> headers = const {},
     bool loadEntities = true,
   }) async {
-    final gridViewUrl = _generateApptiveGridUri(uri);
-
-    final gridHeaders = _createHeadersWithDefaults(headers);
-    gridHeaders.addEntries([ApptiveGridHalVersion.v2.header]);
-    final gridViewResponse =
-        await _client.get(gridViewUrl, headers: gridHeaders);
-    if (gridViewResponse.statusCode >= 400) {
-      if (gridViewResponse.statusCode == 401 && !isRetry) {
-        await authenticator.checkAuthentication();
-        return loadGrid(
-          uri: uri,
-          sorting: sorting,
-          filter: filter,
-          isRetry: true,
-        );
-      }
-      throw gridViewResponse;
-    }
+    final gridViewResponse = await performApptiveLink<http.Response>(
+      link: ApptiveLink(uri: uri, method: 'get'),
+      headers: headers,
+      halVersion: ApptiveGridHalVersion.v2,
+      isRetry: isRetry,
+      parseResponse: (response) async => response,
+    );
 
-    final gridToParse = jsonDecode(gridViewResponse.body);
+    final gridToParse = jsonDecode(gridViewResponse!.body);
     final grid = Grid.fromJson(gridToParse);
```

Comment:
wouldn't it make sense to do these 2 operations in the `parseResponse` field?

Reply:
So in Lines 362-363 we actually do manipulation on the json of the response and the parsing it.
Theoretically we could also instead parse the entities response into GridRows and manipulate the grid however I feel like there is not really a reason to not go with this solution as this is how we always did it for this call to use the response directly.